www/class/config-generate-remote/abstract/host.class.php

```diff
@@ -0,0 +1,275 @@
+            $host['htpl'] = $this->stmt_htpl->fetchAll(PDO::FETCH_COLUMN);
+        }
+
+        $host_template = HostTemplate::getInstance($this->dependencyInjector);
+        $order = 1;
+        foreach ($host['htpl'] as $template_id) {
+            $host_template->generateFromHostId($template_id);
+            hostTemplateRelation::getInstance($this->dependencyInjector)->addRelation($host['host_id'], $template_id, $order);
+            $order++;
+        }
+    }
+
+    protected function getContacts(&$host)
+    {
+        if (!isset($host['contacts_cache'])) {
+            if (is_null($this->stmt_contact)) {
+                $this->stmt_contact = $this->backend_instance->db->prepare("SELECT 
+                    contact_id
+                FROM contact_host_relation
+                WHERE host_host_id = :host_id
+                ");
+            }
+            $this->stmt_contact->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+            $this->stmt_contact->execute();
+            $host['contacts_cache'] = $this->stmt_contact->fetchAll(PDO::FETCH_COLUMN);
+        }
+
+        $contact = Contact::getInstance($this->dependencyInjector);
+        foreach ($host['contacts_cache'] as $contact_id) {
+            $contact->generateFromContactId($contact_id);
+            contactHostRelation::getInstance($this->dependencyInjector)->addRelation($host['host_id'], $contact_id);
+        }
+    }
+
+    protected function getContactGroups(&$host)
+    {
+        if (!isset($host['contact_groups_cache'])) {
+            if (is_null($this->stmt_cg)) {
+                $this->stmt_cg = $this->backend_instance->db->prepare("SELECT 
+                    contactgroup_cg_id
+                FROM contactgroup_host_relation
+                WHERE host_host_id = :host_id
+                ");
+            }
+            $this->stmt_cg->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+            $this->stmt_cg->execute();
+            $host['contact_groups_cache'] = $this->stmt_cg->fetchAll(PDO::FETCH_COLUMN);
+        }
+
+        $cg = Contactgroup::getInstance($this->dependencyInjector);
+        foreach ($host['contact_groups_cache'] as $cg_id) {
+            $cg->generateFromCgId($cg_id);
+            contactgroupHostRelation::getInstance($this->dependencyInjector)->addRelation($host['host_id'], $cg_id);
+        }
+    }
+
+    public function isHostTemplate($host_id, $host_tpl_id)
+    {
+        $loop = array();
+        $stack = array();
+
+        $hosts_tpl = HostTemplate::getInstance($this->dependencyInjector)->hosts;
+        $stack = $this->hosts[$host_id]['htpl'];
+        while (($host_id = array_shift($stack))) {
+            if (isset($loop[$host_id])) {
+                continue;
+            }
+            $loop[$host_id] = 1;
+            if ($host_id == $host_tpl_id) {
+                return 1;
+            }
+            $stack = array_merge($hosts_tpl[$host_id]['htpl'], $stack);
+        }
+
+        return 0;
```

Comment:
```suggestion
        return false;
```

Reply:
(used in an if condition in config-generate/servicetemplate.class.php +133 and config-generate-remote/servicetemplate.class.php +101)